ExtendedLine[] lines
Edge[] edges
Room[] rooms
Contour contour